requirements.txt

```diff
@@ -2,6 +2,7 @@ cython >= 0.29.0
 h5py >= 3.0.0
 numpy >= 1.20.0
 scipy >= 1.3.0
+cupy >= 12.0.0
```

Comment:
We don't want to require cupy by default, so we cannot have cupy in requirements.txt.
So far I think we have just recommended users to install cupy if GPU features are required.
You would have to "skip" testing your code if cupy is not available for the CI test.